flake.nix

```
@@ -0,0 +1,65 @@
+{
+  description = "A flake for the syn data fidelity library";
+
+  inputs = {
+    linux-nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.05";
+    darwin-nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.11";
+  };
+
+  outputs = { self, linux-nixpkgs, darwin-nixpkgs, flake-utils, ... }: let
+    pkgs = {
+      x86_64-linux = linux-nixpkgs.legacyPackages.x86_64-linux;
+      aarch64-darwin = darwin-nixpkgs.legacyPackages.aarch64-darwin;
```

I'm on the last model of Intel Mac, so this doesn't work on my machine, fwiw.

```suggestion
      x86_64-linux = linux-nixpkgs.legacyPackages.x86_64-linux;
      aarch64-darwin = darwin-nixpkgs.legacyPackages.aarch64-darwin;
      x86_64-darwin = darwin-nixpkgs.legacyPackages.x86_64-darwin;
```